src/formatting/dateTime.ts

```diff
@@ -0,0 +1,22 @@
+import { Moment } from 'moment'
+import { currentLanguage } from './locale'
+
+function parseDate (input: string|Moment|Date): Date {
+  if (typeof input === 'string' || input instanceof Date) {
```

Comment:
why doing `new Date` if you already have instance of `Date`?

Reply:
what happens on empty string, number, undefined, null?

Reply:
Date
-empty string returns: Invalid Date
-undefined returns: error Cannot read properties of undefined 
-null returns error Cannot read properties of null

Number
-empty string returns: 0
-undefined returns: NaN
-null returns: 0

Reply:
Added check if date is null or undefined convert to string so that Invalid Date should also be displayed without errors

Reply:
Now creating new Date only if instance is string..